### PR TITLE
added start of Python directories

### PR DESCRIPTION
**Pull request by granrothge**
This is just a start to get what I have now into the wild.

### CODE REVIEW

File: python_tools/MnO_swg_powder_mp.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        in_res.append(genie.getPoints())
+    return in_res
+
+def get_neighbor_idx(m_at,latt,n_order,max_radii=3.):
+    """
+    get the indexes neighbors of order n_order
+    0 is first order
+    """
+    n_at=latt.get_number_of_atoms()
+    num_n=nl.NeighborList(ones(n_at)*max_radii,self_interaction=False,bothways=True)
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder_mp.py]: `np.ones`?

File: python_tools/MnO_swg_powder_mp.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+# use ase to define the lattice.  For spin wave genie one needs to define all
+# atoms in the unit cell.
+mn=sg.crystal(["Mn"],[(0,0,0)], spacegroup=225,
+        cellpar=[a,a,a,90.,90.,90.],size=(2,2,2))
+n_at=mn.get_number_of_atoms()
+# define the magnetic sublattices to be FCC-II antiferromagnetism
+# define 1st three atoms by hand and then loop through all NNN to be sure all
+# spins are defined
+# you can use ase.visualize.view to check if it is correct
+mm=zeros(n_at)
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder_mp.py]: `np.zeros`

File: python_tools/MnO_swg_powder_mp.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+mm[0:3]=spin
+mm[3]=-spin
+at_num=0
+while prod(abs(mm))<1:
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder_mp.py]: `np.prod`

File: python_tools/MnO_swg_powder_mp.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    nnnidxs=get_neighbor_idx(at_num,mn,1,max_radii=3.)
+    mm[nnnidxs]=-mm[at_num]
+    at_num+=1
+mn.set_initial_magnetic_moments(mm)
+inpos=mn.get_scaled_positions()
+#Define spin direction as in the plane
+# component perp to plane
+Sx_vec=cross(inpos[1,:],inpos[2,:])/norm(cross(inpos[1,:],inpos[2,:]))
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder_mp.py]: `np.cross`
`np.linalg.norm`

File: python_tools/MnO_swg_powder_mp.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+# component in plane and perpendicular to moment direction.
+Sy_vec=inpos[1,:]/norm(inpos[1,:])
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder_mp.py]: `np.linalg.norm`

File: python_tools/MnO_swg_powder_mp.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+# Moment direction
+Sz_vec=cross(Sx_vec,Sy_vec)
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder_mp.py]: `np.cross`

File: python_tools/MnO_swg_powder_mp.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+#Mn1.setMoment(2.5,0,0)
+#generate sublattice with - spin direction
+Mn2=swg.Sublattice()
+Mn2.setName("Mn2")
+Mn2.setType("MN2")
+Mn2.setMoment(2.5,1.99133,2.6779)
+# add the subblatices to the cells
+cell.addSublattice(Mn1)
+cell.addSublattice(Mn2)
+atom_pos=zeros(mn.positions.shape)
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder_mp.py]: `np.zeros`

File: python_tools/MnO_swg_powder_mp.py [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+spin_dict={2.5:"Mn1",-2.5:"Mn2"}
+for idx in range(len(mn.positions)):
+    pos=inpos[idx]
+    tlst=list(pos)
+    tlst.insert(0,spin_dict[mn.arrays['magmoms'][idx]])
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder_mp.py:97]: I had to change this to `'initial_magmoms'`. DId ase change the keyword?

File: python_tools/MnO_swg_powder.py [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        in_res.append(genie.getPoints())
+    return in_res
+
+def get_neighbor_idx(m_at,latt,n_order,max_radii=3.):
+    """
+    get the indexes neighbors of order n_order
+    0 is first order
+    """
+    n_at=latt.get_number_of_atoms()
+    num_n=nl.NeighborList(ones(n_at)*max_radii,self_interaction=False,bothways=True)
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder.py]: `np.ones`

File: python_tools/MnO_swg_powder.py [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+# use ase to define the lattice.  For spin wave genie one needs to define all
+# atoms in the unit cell.
+mn=sg.crystal(["Mn"],[(0,0,0)], spacegroup=225,
+        cellpar=[a,a,a,90.,90.,90.],size=(2,2,2))
+n_at=mn.get_number_of_atoms()
+# define the magnetic sublattices to be FCC-II antiferromagnetism
+# define 1st three atoms by hand and then loop through all NNN to be sure all
+# spins are defined
+# you can use ase.visualize.view to check if it is correct
+mm=zeros(n_at)
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder.py]: `np.zeros`

File: python_tools/MnO_swg_powder.py [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+at_num=0
+while prod(abs(mm))<1:
+    nnnidxs=get_neighbor_idx(at_num,mn,1,max_radii=3.)
+    mm[nnnidxs]=-mm[at_num]
+    at_num+=1
+mn.set_initial_magnetic_moments(mm)
+inpos=mn.get_scaled_positions()
+#Define spin direction as in the plane
+# component perp to plane
+Sx_vec=cross(inpos[1,:],inpos[2,:])/norm(cross(inpos[1,:],inpos[2,:]))
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder.py]: `np.cross`
`np.linalg.norm`

File: python_tools/MnO_swg_powder.py [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+# component in plane and perpendicular to moment direction.
+Sy_vec=inpos[1,:]/norm(inpos[1,:])
+# Moment direction
+Sz_vec=cross(Sx_vec,Sy_vec)
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder.py]: `np.cross`

File: python_tools/MnO_swg_powder.py [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+n_at=mn.get_number_of_atoms()
+# define the magnetic sublattices to be FCC-II antiferromagnetism
+# define 1st three atoms by hand and then loop through all NNN to be sure all
+# spins are defined
+# you can use ase.visualize.view to check if it is correct
+mm=zeros(n_at)
+mm[0:3]=spin
+mm[3]=-spin
+at_num=0
+while prod(abs(mm))<1:
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder.py]: `np.prod`

File: python_tools/MnO_swg_powder.py [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+#Mn1.setMoment(2.5,0,0)
+#generate sublattice with - spin direction
+Mn2=swg.Sublattice()
+Mn2.setName("Mn2")
+Mn2.setType("MN2")
+Mn2.setMoment(2.5,1.99133,2.6779)
+# add the subblatices to the cells
+cell.addSublattice(Mn1)
+cell.addSublattice(Mn2)
+atom_pos=zeros(mn.positions.shape)
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder.py]: `np.zeros`

File: python_tools/MnO_swg_powder.py [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+spin_dict={2.5:"Mn1",-2.5:"Mn2"}
+for idx in range(len(mn.positions)):
+    pos=inpos[idx]
+    tlst=list(pos)
+    tlst.insert(0,spin_dict[mn.arrays['magmoms'][idx]])
```
Comment (quantumsteve) [inline python_tools/MnO_swg_powder.py:96]: I had to change this to `'initial_magmoms'`. DId ase change the keyword?

**Review by quantumsteve — CHANGES_REQUESTED** [review]
Minimum changes I had to make to run `MnO_swg_powder.py`

File: python_tools/MnO_swg_powder_mp.py (hunk 1 above)
Comment (granrothge) [reply python_tools/MnO_swg_powder_mp.py]: fixed

File: python_tools/MnO_swg_powder_mp.py (hunk 2 above)
Comment (granrothge) [reply python_tools/MnO_swg_powder_mp.py]: fixed

File: python_tools/MnO_swg_powder_mp.py (hunk 5 above)
Comment (granrothge) [reply python_tools/MnO_swg_powder_mp.py]: fixed

File: python_tools/MnO_swg_powder_mp.py (hunk 6 above)
Comment (granrothge) [reply python_tools/MnO_swg_powder_mp.py]: fixed

File: python_tools/MnO_swg_powder_mp.py (hunk 7 above)
Comment (granrothge) [reply python_tools/MnO_swg_powder_mp.py]: fixed

File: python_tools/MnO_swg_powder_mp.py (hunk 8 above)
Comment (granrothge) [reply python_tools/MnO_swg_powder_mp.py:97]: Hmm yes I believe that is the case, but I will have to look into it some more.

File: python_tools/MnO_swg_powder.py (hunk 15 above)
Comment (granrothge) [reply python_tools/MnO_swg_powder.py:96]: Hmm Yes I believe that is the case but I will have to look into it some more.

**Pull request merged**